src/platforms/hosted/gdb_if.c

```diff
@@ -23,15 +23,15 @@
  * uses a TCP server on port 2000.
  */
 
-#include "general.h"
```

Comment:
It seems this re-ordering blew the build for Linux and may have subtly done so for windows not-cygwin, unsure on fruitos.

Given this, we suggest `#ifndef __CYGWIN__` guarding this version of the include statement, and then `#ifdef __CYGWIN__` guarding the one added below

Reply:
Done, let's see if the build passes now. 😄